internal/controller/pdbwatcher_controller.go

```diff
@@ -161,7 +219,46 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	// What will return us to the original state? Assuming we need a watch on PDB for it to come off
+	// Watch for changes in PDB to revert to original state
+	if pdb.Status.DisruptionsAllowed > 0 && *deployment.Spec.Replicas != pdbWatcher.Status.MinReplicas {
+		// Re-fetch the Deployment to ensure the resource version is up-to-date
```

Comment:
Why would it change between the start and now?

Reply:
you probably don't need this and the check on 231 because if it has changed then an attempt to mutate the previous one  should fail. (because resource version won't match)